doc/man/crystfel.7
.\"
.\" CrystFEL main man page
.\"
.\" Copyright © 2012-2026 Deutsches Elektronen-Synchrotron DESY,
.\"                       a research centre of the Helmholtz Association.
.\"
.\" Part of CrystFEL - crystallography with a FEL
.\"

.TH CRYSTFEL 7
.SH NAME
CrystFEL \- data processing for serial crystallography

.SH DESCRIPTION
CrystFEL is a suite of programs for processing data from serial crystallography
experiments, performed at synchrotron and X-ray free-electron laser facilities,
as well as in your home lab using an electron microscope.

The programs in the suite are:

.IP \fBindexamajig\fR
Batch indexing, integration and data reduction program, which produces a "stream" containing the indexing and integration results for each diffraction pattern.

.IP \fBprocess_hkl\fR
A tool merging intensities from many patterns into a single reflection list, via the Monte Carlo method.

.IP \fBpartialator\fR
Full scaling and post-refinement process for accurate merging of data and outlier rejection.

.IP \fBambigator\fR
A tool for resolving indexing ambiguities.

.IP \fBget_hkl\fR
A tool for manipulating reflection lists, such as performing symmetry expansion.

.IP \fBcell_tool\fR
A tool for manipulating unit cell files, such as applying transformations.

.IP \fBcell_explorer\fR
A tool for examining the distributions of unit cell parameters.

.IP "\fBcompare_hkl\fR and \fBcheck_hkl\fR"
Tools for calculating figures of merit, such as completeness and R-factors.

.IP \fBrender_hkl\fR
A tool for rendering slices of reciprocal space in two dimensions.

.IP \fBalign_detector\fR
A program to refine and optimize detector geometry.

.IP \fBlist_events\fR
A tool for creating lists of events from multi-event data files.

.IP \fBwhirligig\fR
A tool for locating runs of crystals with similar orientations, e.g. from 'mini rotation series' arising from the use of a slow extrusion sample injector.

.IP \fBmake_pixelmap\fR
A tool for creating pixel maps for Cheetah, OnDA etc.

.IP \fBadjust_detector\fR
A tool to alter detector geometry.

.PP
There is also a folder full of scripts for achieving many related tasks.

.PP
CrystFEL mostly works with images stored in HDF5 format, unit cell data in PDB
format, and reflection lists in plain text format (i.e. not MTZ).  There are
scripts for converting both ways between plain text reflection lists and MTZ
files.

.PP
Unit cells can also be given in CrystFEL's own unit cell file format, usually in
a file named \fIsomething.cell\fR.  This can be preferable to a PDB file,
because it states the lattice type and centering explicitly, and because it can
describe a lattice without specifying the cell parameters.  See
\&'man crystfel_unitcell' for the syntax.

.PP
Please see the individual manual pages for the CrystFEL programs for detailed information.

.SH CITING CRYSTFEL
If CrystFEL was important for your research, please consider citing the
following article:
.IP
T. A. White, R. A. Kirian, A. V. Martin, A. Aquila, K. Nass, A. Barty and
H. N. Chapman. "CrystFEL: a software suite for snapshot serial crystallography". J. Appl. Cryst. 45. doi:10.1107/S0021889812002312
.PP
Please let us know (see below) about your publication, so we can include it in
the list of examples on the CrystFEL website.

.SH SYMMETRY IN CRYSTFEL
Without only a very few exceptions, CrystFEL is not interested in space groups.  Instead, it deals with point groups which embody the information about how data should be merged from different crystals.  Every space group belongs to exactly one point group, and you can look up the right one in the International Tables or using the symmetry tables accompanying the CrystFEL source (or to be found on the CrystFEL website in the Theory section).

You can append \fB_ua\fR\fIX\fR to any of the point group symbols listed below, where \fIX\fR is \fBa\fR, \fBb\fR or \fBc\fR.  This will specify the 'unique' axis for lattices which have one.

\fBThe default unique axis, where this makes sense, is c\fR.

Here is a list of all the possible crystallographic point groups, written as they will be accepted by CrystFEL.  Refer
to this list if you experience difficulty with "-y" at any time.

.IP Triclinic
\fB1\fR, \fB-1\fR.

.IP Monoclinic
\fB2/m\fR, \fB2\fR, \fBm\fR.

.IP Orthorhombic
\fBmmm\fR, \fB222\fR, \fBmm2\fR.

.IP Tetragonal
\fB4/m\fR, \fB4\fR, \fB-4\fR, \fB4/mmm\fR, \fB422\fR, \fB-42m\fR, \fB-4m2\fR, \fB4mm\fR.

.IP "Trigonal (rhombohedral axes)"
\fB3_R\fR, \fB-3_R\fR, \fB32_R\fR, \fB3m_R\fR, \fB-3m_R\fR.

.IP "Trigonal (hexagonal axes)"
\fB3_H\fR, \fB-3_H\fR, \fB321_H\fR, \fB312_H\fR, \fB3m1_H\fR, \fB31m_H\fR, \fB-3m1_H\fR, \fB-31m_H\fR.

.IP Hexagonal
\fB6/m\fR, \fB6\fR, \fB-6\fR, \fB6/mmm\fR, \fB622\fR, \fB-62m\fR, \fB-6m2\fR, \fB6mm\fR.

.IP Cubic
\fB23\fR, \fBm-3\fR, \fB432\fR, \fB-43m\fR, \fBm-3m\fR.

.SH FUNDING ACKNOWLEDGEMENTS
Development of CrystFEL is primarily funded by the Helmholtz Association via DESY.

Partial funding for CrystFEL has been provided by:

* The consortium DAPHNE4NFDI in the context of the work of the NFDI e.V. The
  consortium is funded by the Deutsche Forschungsgemeinschaft (DFG, German
  Research Foundation) - project number 460248799 (2022-).

* European Union’s Horizon 2020 research and innovation programme under grant
  agreement No 857641 (ExPaNDS) (2019-2023).

* X-Probe, a project of the European Union's 2020 Research and Innovation
  Program Under the Marie Skłodowska-Curie grant agreement 637295 (2015-2018).

* The BMBF German-Russian Cooperation SyncFELMed, grant 05K14CHA (2014-2017).

* BioStruct-X, a project funded by the Seventh Framework Programme (FP7) of the
  European Commission (2011-2016).

.SH AUTHOR
This page was written by Thomas White and Valerio Mariani.

.SH REPORTING BUGS
Report bugs to <taw@physics.org>, or visit <http://www.desy.de/~twhite/crystfel>.

.SH COPYRIGHT AND DISCLAIMER
.PD 0
Copyright © 2012-2024 Deutsches Elektronen-Synchrotron DESY, a research centre of the Helmholtz Association.
.PD
.PP
Please read the AUTHORS file in the CrystFEL source code distribution for a full list of contributions and contributors.
.P
CrystFEL is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
.P
CrystFEL is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
.P
You should have received a copy of the GNU General Public License along with CrystFEL.  If not, see <http://www.gnu.org/licenses/>.

.SH SEE ALSO
.BR indexamajig (1),
.BR process_hkl (1),
.BR partialator (1),
.BR compare_hkl (1),
.BR check_hkl (1),
.BR render_hkl (1),
.BR get_hkl (1),
.BR cell_tool (1),
.BR whirligig (1),
.BR list_events (1),
.BR make_pixelmap (1),
.BR crystfel_geometry (5),
.BR crystfel_unitcell (5).


NEWS
There is no NEWS.  Check the version control repository...


crystfel.rb
# frozen_string_literal: true

# Suite for processing Serial Femtosecond Crystallography (SFX) data
class Crystfel < Formula
  desc 'Suite for processing Serial Femtosecond Crystallography (SFX) data'
  homepage 'https://www.desy.de/~twhite/crystfel/index.html'

  stable do
    url 'https://www.desy.de/~twhite/crystfel/crystfel-0.12.0.tar.gz'
    sha256 '1ff71a5e1b08760b226addd44e2d50305f49db676dc8407a604227f5ff30581e'
  end

  head do
    url 'https://gitlab.desy.de/thomas.white/crystfel.git'
  end

  depends_on 'bison' => :build
  depends_on 'meson' => :build
  depends_on 'ninja' => :build
  depends_on 'pkg-config' => :build
  if OS.mac? 
      depends_on 'argp-standalone' => :build
  end
  depends_on 'cairo'
  depends_on 'fftw'
  depends_on 'gdk-pixbuf'
  depends_on 'gsl'
  depends_on 'gtk+3'
  depends_on 'hdf5'
  depends_on 'libpng'
  depends_on 'libtiff'
  depends_on 'pango'
  depends_on 'eigen'

  def install
      args = std_meson_args
      args.delete_at(args.index("--wrap-mode=nofallback"))
      puts args
      if OS.mac? 
        args << "-Dc_args=-I/opt/argp-standalone/include"
        args << "-Dcpp_args=-I/opt/argp-standalone/include"
        args << "-Dc_link_args=-L/usr/local/opt/argp-standalone/lib -largp"
        args << "-Dcpp_link_args=-L/usr/local/opt/argp-standalone/lib -largp"
      end
      system 'meson', 'setup', *args, 'build'
      system 'ninja','-C','build'
      system 'ninja','-C','build','test'
      system 'ninja','-C','build','install'
  end

  test do
    # All functionality tests are done in the above section with make test
    system bin / 'ambigator', '--help'
    system bin / 'cell_explorer', '--help'
    system bin / 'cell_tool', '--help'
    system bin / 'check_hkl', '--help'
    system bin / 'compare_hkl', '--help'
    system bin / 'crystfel', '--help'
    system bin / 'align_detector', '--help'
    system bin / 'get_hkl', '--help'
    system bin / 'list_events', '--help'
    system bin / 'make_pixelmap', '--help'
    system bin / 'partialator', '--help'
    system bin / 'process_hkl', '--help'
    system bin / 'render_hkl', '--help'
    system bin / 'whirligig', '--help'
    # indexamajig --help crashes on MacOS due to argp-standalone bug
  end
end
